cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -0,0 +1,67 @@
+"""Tests for Openstack cloud Flavors"""
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([OpenStackProvider], scope='function')
+]
+
+
+DISK_SIZE = 1
+RAM = 64
+VCPUS = 1
+SWAP_SIZE = 1
+RXTX = 1
+ZERO_DISK_SIZE = 0
+
+
+@pytest.fixture(scope='function')
+def flavor(appliance, provider):
+    collection = appliance.collections.cloud_flavors
+    flavor = collection.create(name=fauxfactory.gen_alpha(),
+                               provider=provider,
+                               ram=RAM,
+                               vcpus=VCPUS,
+                               disk=DISK_SIZE,
+                               swap=SWAP_SIZE,
+                               rxtx=RXTX)
+    yield flavor
+
+    if flavor.exists:
+        flavor.delete(wait=False)
+
+
+@pytest.fixture(scope='function')
+def zero_disk_flavor(provider):
+    api_flv = provider.mgmt.api.flavors.create(fauxfactory.gen_alpha(), RAM, VCPUS, ZERO_DISK_SIZE)
+    wait_for(provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
+    zero_disk_flavor = provider.appliance.collections.cloud_flavors.instantiate(
+        api_flv.name, provider)
+
+    yield zero_disk_flavor
+
+    if zero_disk_flavor.exists:
+        zero_disk_flavor.delete(wait=False)
+
+
+def test_create_flavor(flavor):
+    assert flavor.exists
+
+
+def test_delete_flavor(flavor):
+    flavor.delete()
+    assert not flavor.exists
+
+
+def test_create_zero_disk_flavor(zero_disk_flavor):
```

Comment:
I don't think this is a valid creation test, as the fixture is creating the flavor on the provider directly via its API, which has nothing to do with MIQ.

This is really a test of provider refresh with a zero_disk_flavor, and a test of deleting through UI.

Again this should all be one test, not a separate create/delete test pair.